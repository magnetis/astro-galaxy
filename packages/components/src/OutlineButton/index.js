import styled from 'styled-components';
import Button from '../Button';

const hoverStyles = props => ({
  'outline.uranus': `
    background-color: ${props.theme.colors.uranus400};
    border-color: ${props.theme.colors.uranus400};
    color: ${props.theme.colors.space100};
  `,
  'outline.earth': `
    background-color: ${props.theme.colors.earth300};
    border-color: ${props.theme.colors.earth300};
    color: ${props.theme.colors.moon900};
  `,
  'outline.venus': `
    background-color: ${props.theme.colors.venus300};
    border-color: ${props.theme.colors.venus300};
    color: ${props.theme.colors.space100};
  `,
  'outline.mars': `
    background-color: ${props.theme.colors.mars400};
    border-color: ${props.theme.colors.mars400};
    color: ${props.theme.colors.space100};
  `,
});

const activeStyles = props => ({
  'outline.uranus': `
    background-color: ${props.theme.colors.uranus600};
    border-color: ${props.theme.colors.uranus600};
  `,
  'outline.earth': `
    background-color: ${props.theme.colors.earth500};
    border-color: ${props.theme.colors.earth500};
  `,
  'outline.venus': `
    background-color: ${props.theme.colors.venus500};
    border-color: ${props.theme.colors.venus500};
  `,
  'outline.mars': `
    background-color: ${props.theme.colors.mars600};
    border-color: ${props.theme.colors.mars600};
  `,
});

const OutlineButton = styled(Button)`
  background-color: transparent;

  :disabled {
    cursor: not-allowed;
    background-color: transparent;
    border-color: ${props => props.theme.colors.moon200};
    color: ${props => props.theme.colors.moon200};
  }

  :hover:not(:disabled),
  :focus:not(:disabled) {
    ${props => hoverStyles(props)[props.variant]}
  }

  :active:not(:disabled) {
    ${props => activeStyles(props)[props.variant]}
  }

  :focus {
    outline: none;
  }
`;

OutlineButton.defaultProps = {
  variant: 'outline.uranus',
  buttonSize: 'medium',
};

export default OutlineButton;
