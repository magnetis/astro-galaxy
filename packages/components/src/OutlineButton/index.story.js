import React from 'react';
import { storiesOf } from '@storybook/react';
import OutlineButton from './index.js';

storiesOf('outline buttons', module)
  .add('button default', () => <OutlineButton>uranus</OutlineButton>)
  .add('outline buttons variants', () => (
    <React.Fragment>
      <OutlineButton variant="outline.uranus">uranus</OutlineButton>
      <OutlineButton variant="outline.earth">earth</OutlineButton>
      <OutlineButton variant="outline.venus">venus</OutlineButton>
      <OutlineButton variant="outline.mars">mars</OutlineButton>
      <OutlineButton disabled>disabled</OutlineButton>
    </React.Fragment>
  ));

storiesOf('outline button sizes', module).add('outline buttons sizes', () => (
  <React.Fragment>
    <OutlineButton buttonSize="small">small</OutlineButton>
    <OutlineButton buttonSize="medium">medium ~ default</OutlineButton>
    <OutlineButton buttonSize="large">large</OutlineButton>
  </React.Fragment>
));
