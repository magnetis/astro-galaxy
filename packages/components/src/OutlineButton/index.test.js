import React from 'react';
import toJSON from 'enzyme-to-json';
import { mountWithTheme } from '@magnetis/astro-galaxy-core';
import OutlineButton from './index';

describe('OutlineButton', () => {
  it('should render a OutlineButton component', () => {
    const wrapper = mountWithTheme(<OutlineButton>uranus</OutlineButton>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
