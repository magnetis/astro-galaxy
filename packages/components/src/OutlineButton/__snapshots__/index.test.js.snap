// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton should render a OutlineButton component 1`] = `
<OutlineButton
  buttonSize="medium"
  display="inline-block"
  font="primary"
  variant="outline.uranus"
>
  <StyledComponent
    buttonSize="medium"
    display="inline-block"
    font="primary"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "_foldedDefaultProps": Object {
          "buttonSize": "medium",
          "display": "inline-block",
          "font": "primary",
          "variant": "outline.uranus",
        },
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "OutlineButton-sc-9jysz0-0",
          "isStatic": false,
          "lastClassName": "jakqGW",
          "rules": Array [
            "cursor:pointer;white-space:no-wrap;text-align:center;transition:all 0.3s ease;overflow:hidden;border-width:2px;border-style:solid;:disabled{cursor:not-allowed;background-color:",
            [Function],
            ";border-color:",
            [Function],
            ";color:",
            [Function],
            ";}:hover:not(:disabled),:focus:not(:disabled){",
            [Function],
            "}:active:not(:disabled){",
            [Function],
            "}:focus{outline:none;}",
            [Function],
            ";",
            [Function],
            ";",
            [Function],
            ";",
            [Function],
            ";",
            [Function],
            ";",
            [Function],
            ";",
            [Function],
            ";",
            [Function],
            ";",
            [Function],
            ";",
            "background-color:transparent;:disabled{cursor:not-allowed;background-color:transparent;border-color:",
            [Function],
            ";color:",
            [Function],
            ";}:hover:not(:disabled),:focus:not(:disabled){",
            [Function],
            "}:active:not(:disabled){",
            [Function],
            "}:focus{outline:none;}",
          ],
        },
        "displayName": "OutlineButton",
        "foldedComponentIds": Array [
          "Button-tgj28e-0",
        ],
        "propTypes": Object {},
        "render": [Function],
        "styledComponentId": "OutlineButton-sc-9jysz0-0",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    variant="outline.uranus"
  >
    <button
      className="Button-tgj28e-0 OutlineButton-sc-9jysz0-0 jakqGW"
      display="inline-block"
    >
      uranus
    </button>
  </StyledComponent>
</OutlineButton>
`;
